src/components/connect.js

```diff
@@ -192,7 +196,14 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         this._unmounted = true
       }
 
+      shouldComponentUpdate(nextProps, nextState) {
+        return !options.pure ||
+          this.state.data != nextState.data ||
+          (dependsOnProps ? (this._propsDidChange || this.props.children != nextProps.children) : !shallowEqual(this.props, nextProps))
```

Comment:
should i add a comment explaining `this._propsDidChange` is defined in `componentWillReceiveProps` and will be used in `componentShouldUpdate`?